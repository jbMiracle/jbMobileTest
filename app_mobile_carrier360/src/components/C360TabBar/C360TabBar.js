import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Col, Grid, Row, connectStyle } from 'native-base';
import { SafeAreaView, View, Platform } from 'react-native';
import { styleDef } from './C360TabBar.style';

class C360TabBar extends PureComponent {
  render() {
    const { navigation, renderIcon, style } = this.props;
    const { routes } = navigation.state;

    if (Platform.OS === 'ios') {
      return (
        <SafeAreaView style={ [style.safeView, style.safeViewIos] }>
          <View style={ style.outerContainer }>
            <View>
              <View style={ style.bar }>
                <View style={ style.mainButtonHump } />
              </View>
              <View style={ style.topBar } />
            </View>
            <Grid style={ style.contents }>
              <Row>
                { renderTabs({ navigation, renderIcon, style }, routes) }
              </Row>
            </Grid>
          </View>
        </SafeAreaView>
      );
    }

    return (
      <View style={ style.safeView }>
        <View style={ style.bar } />
        <View style={ style.mainButtonHumpAndroid } />
        <View style={ style.topAndroidBar } />
        <Grid style={ style.contents }>
          <Row>
            { renderTabs(this.props, routes) }
          </Row>
        </Grid>
      </View>
    );
  }
}

function renderTabs({ navigation, renderIcon, style }, allRoutes) {
  return allRoutes.map((route, index) => {
    const focused = index === navigation.state.index;
    const scene = { route, index, focused };
    const columnStyle = (Platform.OS === 'ios') ? null : style.col;

    return (
      <Col key={ route.key } style={ columnStyle }>
        { renderIcon(scene) }
      </Col>
    );
  });
}

C360TabBar.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  renderIcon: PropTypes.func.isRequired,
  style: PropTypes.object.isRequired,
};

export default connectStyle('C360.TabBar', styleDef)(C360TabBar);
