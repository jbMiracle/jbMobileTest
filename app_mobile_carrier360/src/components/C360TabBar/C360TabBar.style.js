import { Platform } from 'react-native';
import C360Colors from '../../themes/360Colors';

export const tabBarHeight = 60;
export const overallBarHeight = Platform.OS === 'ios' ? 60 : 76;

export const styleDef = {
  outerContainer: {
    height: tabBarHeight,
  },
  bar: {
    backgroundColor: C360Colors.White,
    height: tabBarHeight,
    shadowOffset: { width: 0, height: -3 },
    shadowRadius: 5,
    shadowOpacity: 0.1,
  },
  topBar: {
    backgroundColor: C360Colors.White,
    height: tabBarHeight,
    top: 1,
    width: '100%',
    position: 'absolute',
  },
  topAndroidBar: {
    backgroundColor: C360Colors.White,
    height: tabBarHeight,
    bottom: -1,
    width: '100%',
    position: 'absolute',
  },
  mainButtonHump: {
    alignSelf: 'center',
    backgroundColor: C360Colors.White,
    height: tabBarHeight,
    width: tabBarHeight + 3,
    marginTop: -20,
    borderRadius: tabBarHeight / 2,
    shadowOffset: { width: 0, height: -3 },
    shadowRadius: 8,
    shadowOpacity: 0.1,
    position: 'absolute',
  },
  mainButtonHumpAndroid: {
    alignSelf: 'center',
    backgroundColor: C360Colors.White,
    height: tabBarHeight,
    width: tabBarHeight + 3,
    bottom: 16,
    borderRadius: tabBarHeight / 2,
    shadowOffset: { width: 0, height: -3 },
    shadowRadius: 8,
    shadowOpacity: 0.1,
    position: 'absolute',
  },
  contents: {
    backgroundColor: 'transparent',
    position: 'absolute',
    width: '100%',
    height: overallBarHeight,
  },
  safeView: {
    backgroundColor: 'transparent',
    justifyContent: 'flex-end',
    height: overallBarHeight,
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    marginTop: overallBarHeight,
  },
  safeViewIos: {
    backgroundColor: C360Colors.White,
  },
  col: {
    justifyContent: 'flex-end',
  },
};
