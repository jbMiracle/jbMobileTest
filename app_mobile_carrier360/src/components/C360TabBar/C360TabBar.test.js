import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360TabBar from './C360TabBar';

it('can render tab bar', () => {
  const renderer = new ShallowRenderer();
  const props = {
    navigation: {
      state: {
        routes: {
          map: () => {},
        },
        index: 1,
      },
      navigate: () => {},
      getParam: () => {},
    },
    renderIcon: () => {},
  };
  renderer.render(<C360TabBar { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
