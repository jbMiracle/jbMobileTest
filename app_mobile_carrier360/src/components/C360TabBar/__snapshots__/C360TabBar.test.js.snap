// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render tab bar 1`] = `
<C360TabBar
  navigation={
    Object {
      "getParam": [Function],
      "navigate": [Function],
      "state": Object {
        "index": 1,
        "routes": Object {
          "map": [Function],
        },
      },
    }
  }
  renderIcon={[Function]}
  style={
    Object {
      "bar": Object {
        "backgroundColor": "#FFFFFF",
        "height": 60,
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      },
      "col": Object {
        "justifyContent": "flex-end",
      },
      "contents": Object {
        "backgroundColor": "transparent",
        "height": 60,
        "position": "absolute",
        "width": "100%",
      },
      "mainButtonHump": Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 30,
        "height": 60,
        "marginTop": -20,
        "position": "absolute",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
        "width": 63,
      },
      "mainButtonHumpAndroid": Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 30,
        "bottom": 16,
        "height": 60,
        "position": "absolute",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
        "width": 63,
      },
      "outerContainer": Object {
        "height": 60,
      },
      "safeView": Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "height": 60,
        "justifyContent": "flex-end",
        "left": 0,
        "marginTop": 60,
        "position": "absolute",
        "right": 0,
      },
      "safeViewIos": Object {
        "backgroundColor": "#FFFFFF",
      },
      "topAndroidBar": Object {
        "backgroundColor": "#FFFFFF",
        "bottom": -1,
        "height": 60,
        "position": "absolute",
        "width": "100%",
      },
      "topBar": Object {
        "backgroundColor": "#FFFFFF",
        "height": 60,
        "position": "absolute",
        "top": 1,
        "width": "100%",
      },
    }
  }
  virtual={undefined}
/>
`;
